sync.go

```diff
@@ -132,7 +132,11 @@ func (w *PipeWriter) Write(p []byte) (n int, err error) {
 			}
 
 			// chunk write to fill space
-			m, err = w.b.Write(p[n : int64(n)+gap(w.b)])
```

Comment:
Looks good, merged it in and included it in release v2.0.1.

I'll probably try and add some more tests to hit this case/boost coverage. It's hard to test toctou conditions reliably. Kicking myself that I missed this, but thankful you caught it!